torch/testing/_internal/common_methods_invocations.py

```diff
@@ -2518,22 +2593,38 @@ def gradcheck_wrapper_triangular_input(op, input, *args, upper=False, **kwargs):
                        SkipInfo('TestGradients', 'test_method_grad',
                                 device_type='cuda', dtypes=[torch.cdouble], active_if=IS_WINDOWS),
                    )),
+    OpInfo('add',
+           dtypes=all_types_and_complex_and(torch.bool, torch.bfloat16, torch.float16),
+           assert_autodiffed=True,
+           sample_inputs_func=sample_inputs_add,
+           supports_inplace_autograd=False),
     OpInfo('addmm',
-           dtypes=floating_types(),
+           variant_test_name='without_non_fusible_nodes',
+           dtypes=floating_types_and(torch.float16),
            dtypesIfCPU=all_types_and_complex_and(torch.float16, torch.bfloat16),
-           # BFloat16 support on CUDA requires CUDA 11 and SM53
-           dtypesIfCUDA=floating_types_and(torch.float16, torch.complex64, torch.complex128,
-                                           *[torch.bfloat16] if CUDA11OrLater else []),
-           dtypesIfROCM=floating_types_and(torch.half),
+           dtypesIfCUDA=floating_and_complex_types_and(torch.float16, *[torch.bfloat16] if CUDA11OrLater else []),
            assert_autodiffed=True,
-           autodiff_nonfusible_nodes=['aten::add', 'aten::mm'],
+           supports_inplace_autograd=False,
            skips=(
-               # Skips unsupported bfloat16 check because above support check
-               #   doesn't work on all platforms
-               SkipInfo('TestOpInfo', 'test_unsupported_dtypes', dtypes=(torch.bfloat16,)),
-               # TODO: remove redundant method_tests() entries
-               SkipInfo('TestOpInfo', 'test_duplicate_method_tests')),
+               # Skips unsupported bfloat16 check because cuDNN 8 & below
+               # are buggy. This test works on cuDNN 8.1 & CUDA 11.2
+               SkipInfo('TestOpInfo', 'test_supported_backward', dtypes=[torch.bfloat16], device_type='cuda'),
+           ),
```

Comment:
Hello @anjali411 & @mruberry, due to the CUDA bug that has been fixed in CUDA 11.2 & cuDNN 8.1, I've enabled `bfloat16` for `addmm` & `mm` on CUDA 11+, but am skipping `test_supported_backward` for it, because CI tests run on CUDA 10.2 & CUDA 11.1 (with older versions of cuDNN). If you'd agree with this approach, maybe it can be done for `addbmm` in #55649 as well. This PR is ready for review. Thank you!

Reply:
Instead of skipping the tests based upon my speculation, I'm running experiments to confirm my hypothesis.

Reply:
`addmm` & `mm`'s `test_supported_backward` are passing on CUDA11 without checking for CUDA SM capability

So, now I know they're not related to `addbmm`, whose corresponding test fails (and will fail) on CUDA 11.1.
I'll unskip these tests for `addmm` & `mm`.

Reply:
The GPUs used by CircleCI for linux tests are actually SM_52

Reply:
Thank you for investigating, @imaginary-person; your attention to detail is impressive.